src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java

```diff
@@ -333,6 +333,20 @@ public void safeRemove(PhysicalLocation entity) {
 		return convert(locations);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<PhysicalLocation> findLocationByIdWithChildren(boolean returnGeometry, String id) {
```

Comment:
Add limit as a param to this method so that its possible to send a limit from the API